lib/buffer.js

```diff
@@ -146,9 +143,8 @@ function Buffer(arg, encodingOrOffset, length) {
   // Common case.
   if (typeof arg === 'number') {
     if (typeof encodingOrOffset === 'string') {
-      throw new Error(
-        'If encoding is specified then the first argument must be a string'
-      );
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'first argument',
```

Comment:
Since this is checking for the https://nodejs.org/api/buffer.html#buffer_new_buffer_string_encoding case `first argument` -> `string`